Redazione iniziale del documento Norme di Progetto (#40)
SevenBitsSwe/7BitsDocs: RTB/Norme_di_progetto.tex

            \begin{figure}[H]
            \centering
            \includegraphics[width=0.5\textwidth]{AssociazioneClasse.png}
            \caption{Rappresentazione della relazione di associazione.}
            \end{figure}
        
            \item \textbf{Generalizzazione}: La relazione di generalizzazione si verifica quando una classe eredita attributi, comportamenti e relazioni dalla classe genitore. Nello specifico ogni istanza della classe figlia è anche un'istanza della classe genitore, ma con caratteristiche o dettagli aggiuntivi specifici. La generalizzazione è rappresentata graficamente con una freccia vuota che collega la classe figlia (B) alla classe genitore (A);
            \begin{figure}[H]
            \centering
            \includegraphics[width=0.15\textwidth]{GeneralizzazioneClasse.png}
            \caption{Rappresentazione della relazione di generalizzazione.}
            \end{figure}
        \end{itemize}
    
        \paragraph{Metriche}
        %da completare e inserire tabella%

    \subsubsection{Codifica}

>         \paragraph{Descrizione e scopo} 

Reviewer: ```suggestion
        \paragraph{Descrizione e Scopo} 
```

            \item YYYY: anno con 4 cifre;
            \item MM: mese con 2 cifre;
            \item DD: giorno con 2 cifre;
        \end{itemize}

    \subsubsection{Strumenti}
    Il gruppo ha deciso di utilizzare i seguenti strumenti:
    \begin{itemize}
        \item \textbf{Git}: Strumento per il controllo di versione distribuito impiegato dal gruppo per gestire i documenti e le versioni successive di essi prodotte in maniera asincrona dai membri;
        \item \textbf{GitHub}: Piattaforma di versionamento e repository per la documentazione. Permette la gestione del codice sorgente, il controllo delle modifiche tramite funzionalità come le issue e le pull request (PR), facilitando la collaborazione all’interno di un team di sviluppo;
        \item \textbf{\LaTeX}: Linguaggio scelto per la redazione dei documenti, spesso utilizzato in ambito accademico, scientifico e tecnico;
        \item \textbf{Overleaf}: Servizio utilizzato dal gruppo per la creazione e modifica sincrona da parte di due o più componenti di file .tex usati per produrre documentazione;
    \end{itemize}

    \subsection{Gestione della configurazione}

    \subsection{Verifica}
    \label{verifica}

>     \subsubsection{Descrizione}

Reviewer: ```suggestion
    \subsubsection{Descrizione e Scopo}
```

>     La verifica è un processo essenziale che deve essere applicato a tutti gli altri processi per poterli considerare completati. Il suo scopo principale è quello di assicurarsi che i prodotti siano corretti ed aderiscano ai vincoli individuati ed elencati nel documento \textit{Piano di Qualifica}. Le attività di verifica sono assegnate ai verificatori, che si impegnano a far rispettare i vincoli di qualità, presenti nel \textit{Piano di Qualifica}, correggendo e/o segnalando quando sono presenti modifiche non conformi agli standard.

Reviewer: ```suggestion
    La verifica è un processo essenziale che deve essere applicato a tutti gli altri processi per poterli considerare completati. Il suo scopo principale è quello di assicurare che i prodotti siano corretti e che aderiscano ai vincoli individuati ed elencati nel documento \textit{Piano di Qualifica}. Le attività di verifica sono assegnate ai verificatori, che si impegnano a far rispettare i vincoli di qualità presenti nel \textit{Piano di Qualifica}, correggendo e/o segnalando qualora siano presenti contenuti non conformi agli standard.
```

    \subsubsection{Strumenti}
    Gli strumenti utilizzati per agevolare il processo di verifica sono i seguenti:
    \paragraph{GitHub}

>     GitHub fornisce un servizio di review all'interno del meccanismo di PR, permettendo al verificatore di visualizzare le ultime modifiche. Il verificatore una volta verificato il documento può inserire commenti specifici, con all'interno correzioni o migliorie da apportare, inviando una richiesta di modifica nella PR o approvare la PR e procedere con la verifica di altri prodotti presenti in repository in attesa di verifica. La review di GitHub consente all'intero team di poter tracciare le modifiche effettuate all'interno della repository.\\

Reviewer: ```suggestion
    GitHub fornisce un servizio di review all'interno del meccanismo di PR, permettendo al verificatore di visualizzare le ultime modifiche. Il verificatore, una volta verificato il documento, può inserire commenti specifici contenenti correzioni o migliorie da apportare inviando una richiesta di modifica nella PR, oppure può approvare la PR e procedere con la verifica di altri prodotti presenti in repository in attesa di verifica. Una volta completata la verifica del documento, il verificatore può inserire commenti specifici con correzioni o suggerimenti per migliorie e inviare una richiesta di modifica nella Pull Request (PR). In alternativa, può approvare la PR e passare all'eventuale verifica di altri prodotti in attesa di revisione presenti nel repository. La review di GitHub consente all'intero team di poter tracciare le modifiche effettuate all'interno della repository.\\
```

    Possono essere assegnate regole alle PR di GitHub per accertarsi che avvenga la verifica di almeno 2 verificatori per poter effettuare la merge nel branch main.
    
    \subsubsection{Analisi statica}

>     Effettuata sul prodotto software o sulla documentazione senza dover eseguire il sistema software. Serve ad accertare la conformità a regole adottate, l'assenza di difett e la presenza di proprietà desiderate, garantendo un livello ottimale di qualità. Le due principali tecniche sono Inspection e Walkthrough. Il gruppo preferisce l'approccio ad Inspection, così da ridurre il tempo di verifica e poter lasciare concentrare l'autore sulla stesura di altri documenti o codice.

Reviewer: ```suggestion
    L'analisi statica è un tipo di analisi che viene effettuata sul prodotto software o sulla documentazione senza dover eseguire il sistema software. Serve ad accertare la conformità con le regole adottate, verificare l'assenza di difetti e la presenza di proprietà desiderate, garantendo un livello ottimale di qualità. Le due principali tecniche sono Inspection e Walkthrough. Il gruppo preferisce l'approccio ad Inspection, così da ridurre il tempo di verifica affinché l'autore possa dedicarsi alla stesura di altri documenti o codice.
```

    \paragraph{Inspection}
    Il verificatore segue una sequenza di passaggi prestabiliti per verificare che si stiano rispettando le regole presenti nel \textit{Piano di Qualifica}
    \paragraph{Walkthrough}

>     Il verificatore e l'autore del codice/documento da verificare discutono sulle modifiche da apportare. Questo processo richiede la presenza e la collaborazione di entrambi i ruoli.Viene effettuato un controllo ad ampio spettro dell'oggetto, senza avere un'idea precisa di cosa andare a verificare.

Reviewer: ```suggestion
    Il verificatore e l'autore del prodotto da verificare discutono sulle modifiche da apportare. Questo processo richiede la presenza sincrona e la collaborazione di entrambi i ruoli, dunque risulta più complessa dell'Inspection in termini organizzativi. Viene effettuato un controllo ad ampio spettro dell'oggetto, senza avere un'idea precisa di cosa andare a verificare.
```

    

    \subsubsection{Analisi dinamica}

>     Effettuata esclusivamente sul prodotto software tramite l'esecuzione effettiva di esso. L'analisi dinamica si divide in test. Ogni test è una esecuzione di un programma e provano il comportamento di singole parti di codice su un insieme finito di casi di prova.\\

Reviewer: ```suggestion
     L'analisi dinamica viene condotta direttamente sul prodotto software attraverso la sua esecuzione effettiva. Questo tipo di analisi si suddivide in diverse tipologie di test, ciascuna finalizzata a verificare aspetti specifici del funzionamento del software. Ogni test corrisponde all'esecuzione di un programma, e simula il comportamento di singole parti di codice su un insieme finito di casi di prova.\\
```

>     Questa procedura misura la qualità del prodotto e assicura che esegua correttamente il suo scopo. I test effettuati devono essere ripetibili nel tempo per assicurare funzionamento persistente durante tutto lo svolgimento del prodotto software.

Reviewer: ```suggestion
    Questa procedura misura la qualità del prodotto e assicura che esso raggiunga correttamente il suo scopo. I test effettuati devono essere ripetibili nel tempo per assicurare un funzionamento persistente durante tutto lo svolgimento del prodotto software.
```

    \subsubsection{Classificazione dei test}
    Tutti i test che verranno eseguiti sono presenti nel documento \textit{Piano di Qualifica} ed il Verificatore è tenuto ad eseguirli riportandone gli esiti all'interno del \textit{Piano di Qualifica}.\\
    Ad ogni test devono essere attribuiti:
    \begin{itemize}

>     \item \textbf{Stato iniziale}: Parametri del software al momento dell'esecuzione del test;

Reviewer: ```suggestion
    \item \textbf{Stato iniziale}: parametri del software al momento dell'esecuzione del test;
```

    \item \textbf{Input}: dati forniti in entrata per l'esecuzione del test;

>     \item \textbf{Output}: risultati attesi in uscita dato uno specifico input.

Reviewer: ```suggestion
    \item \textbf{Output}: risultati attesi in uscita, dato uno specifico input.
```

    \end{itemize}

>     Inoltre i test sono divisi in varie categorie:

Reviewer: Toglierei la lista visto che dopo vengono descritti uno per paragrafo, così risulta coerente a come hai fatto prima per gli strumenti di verifica

    \begin{itemize}
    \item Test di unità;
    \item Test di integrazione;
    \item Test di sistema;
    \item Test di regressione;
    \item Test di accettazione.
    \end{itemize}

    \paragraph{Test di unità}

>     Test eseguiti su singole unità del software, come funzioni o metodi, in modo indipendente dal resto del sistema. L'obiettivo è assicurare che ogni unità di codice funzioni correttamente.Devono essere eseguiti per primi visto che le unità garantiscono il corretto funzionamento dei singoli metodi che poi verranno integrati tra di loro.\\

Reviewer: ```suggestion
    Test eseguiti su singole unità del software, come funzioni o metodi, in modo indipendente dal resto del sistema. L'obiettivo è assicurare che ogni unità di codice funzioni correttamente. Devono essere eseguiti per primi visto che le unità garantiscono il corretto funzionamento dei singoli metodi che poi verranno integrati tra di loro.\\
```

    Sono presenti due tipi di test di unità:
    \begin{itemize}

>     \item \textbf{Funzionale}: verificano che ogni unità esegue le funzione specificate nel design, facendo riferimento solo alla specifica input-output dell'oggetto di verifica;

Reviewer: ```suggestion
    \item \textbf{Funzionale}: verificano che ogni unità esegua le funzioni specificate nel design, facendo riferimento solo alla specifica input-output dell'oggetto di verifica;
```

    \item \textbf{Strutturale}: verificano la logica interna dell'oggetto di verifica esaminando il codice sorgente dell'unità.
    \end{itemize}

    \paragraph{Test di integrazione}
    Test eseguiti per controllare la corretta interazione tra le varie unità del software. Questo tipo di test vengono eseguiti dopo la conferma che tutti i test di unità sono passati con successo.\\
    Vi sono due tipi di integrazione:
    \begin{itemize}
      \item \textbf{Bottom-up}: l'integrazione parte dalle componenti di sistema con minori dipendenze e un maggiore valore interno. Richiede pochi mock ma ritarda la messa a disposizione di funzionalità visibili all'utente.
      \item \textbf{Top-down}: l'integrazione parte dalle componenti di sistema con maggiori dipendenze e un maggiore valore esterno. Richiede molti mock ma integra prima le funzionalità visibili all'utente.
    \end{itemize}
    
    \paragraph{Test di sistema}
    Test eseguiti per controllare che i requisiti specificati nell'Analisi dei requisiti siano rispettati, garantendo che tutte le componenti siano integrate correttamente.\\

>     L'obiettivo di questo test è di accertarsi che l'intero sistema lavori come entità unica, verificando che tutte le singole componenti funzionino insieme. Viene eseguita al completamento dei test di integrazione.

Reviewer: ```suggestion
    L'obiettivo di questo test è di accertarsi che l'intero sistema lavori come entità unica, verificando che tutte le singole componenti funzionino insieme. Viene eseguito al completamento dei test di integrazione.
```

    \paragraph{Test di regressione}
    Garantiscono che le modifiche apportate al codice non introducano nuovi difetti o danneggino il sistema precedentemente funzionante.\\

>     Consiste nella ripetizione selettiva delle tipologie dei test elencati precedentemente e devono essere eseguiti ogni volta che si effettuano modifiche al codice.

Reviewer: ```suggestion
    Consistono nella ripetizione selettiva delle tipologie dei test elencati precedentemente e devono essere eseguiti ogni volta che si effettuano modifiche al codice.
```

    \paragraph{Test di accettazione}

>     Test da passare prima del rilascio del software. Serve a garantire che il prodotto finale soddisfi le aspettative del proponente e degli utenti finali.

Reviewer: ```suggestion
    Sono gli ultimi test da passare prima del rilascio del software. Servono a garantire che il prodotto finale soddisfi le aspettative del proponente e degli utenti finali.
```

    \paragraph{Classificazione dei test}

>     I test vengono identificati in base alla loro tipologia e tramite un codice numerico, specifico all'interno della categoria. I test devono avere la seguente forma: \textbf{T[Tipologia Test][Codice]}\\

Reviewer: ```suggestion
    I test vengono identificati in base alla loro tipologia e tramite un codice numerico, specifico all'interno della categoria. I test devono avere la seguente forma:  
    \begin{quote}
    \textbf{T[Tipologia][Codice]}
    \end{quote}\\
```

>     Tipologia:

Reviewer: ```suggestion
Dove:
                \item [-] \textbf{Tipologia}:
            \begin{itemize}
                \item [*] \textbf{U}: Unità;
                \item [*] \textbf{I}: Integrazione;
                \item [*] \textbf{S}: Sistema;
                \item [*] \textbf{A}: Accettazione;
            \end{itemize}
```

>     \begin{itemize}

Reviewer: Lista da togliere se viene aggiunta sopra. Ho uniformato stilisticamente questa sezione come la 2.2.2.4

        \item \textbf{GitHub}: Piattaforma di versionamento e repository per la documentazione. Permette la gestione del codice sorgente, il controllo delle modifiche tramite funzionalità come le issue e le pull request (PR), facilitando la collaborazione all’interno di un team di sviluppo;
        \item \textbf{\LaTeX}: Linguaggio scelto per la redazione dei documenti, spesso utilizzato in ambito accademico, scientifico e tecnico;
        \item \textbf{Overleaf}: Servizio utilizzato dal gruppo per la creazione e modifica sincrona da parte di due o più componenti di file .tex usati per produrre documentazione;
    \end{itemize}

    \subsection{Gestione della configurazione}
    \label{gestione-configurazione}
    \subsubsection{Descrizione e Scopo}
    Il processo di gestione della configurazione identifica le norme che sono state adottate durante lo svolgimento del progetto al fine di garantire la tracciabilità, la coerenza e la comprensibilità del codice e della documentazione prodotta dal gruppo.

>     Il suo scopo è quello di gestire e organizzare ogni modifica sul codice o sulla documentazione supportando con delle motivazioni alla base dei cambiamenti effettuati e con gli autori relativi

Reviewer 2: ```suggestion
    Il suo scopo è quello di gestire e organizzare ogni modifica sul codice o sulla documentazione supportando con delle motivazioni alla base dei cambiamenti effettuati e con i relativi autori.
```

        \paragraph{Riunioni esterne}
        Le riunioni esterne coinvolgono i membri del gruppo, la proponente e i committenti. Gli incontri con la proponente si terranno sulla piattaforma Google Meet, con l'indirizzo comunicato al gruppo dall'azienda. I membri del gruppo si impegnano a partecipare attivamente, cercando di garantire la loro presenza. In caso di impossibilità a partecipare, ogni membro potrà recuperare la sintesi delle discussioni e le decisioni prese leggendo il verbale della riunione.

    \subsection{Miglioramento}

    \subsection{Formazione}

\newpage
\section{Standard ISO/IEC:9126}
\label{standard_qualità}
Lo standard ISO/IEC:9126 individua una serie di linee guida e normative, sviluppate dall'ISO (Organizzazione 
internazionale per la normazione) in collaborazione con l'IEC (Commissione Elettrotecnica Internazionale), finalizzate 
a descrivere un modello univoco e globalmente riconosciuto di qualità del software. Il modello nasce dalla necessità di

> fornire alle società di software un'approccio organizzato che faciliti e migliori la produzione di software e conseguentemente 

Reviewer 2: ```suggestion
fornire alle società di software un approccio organizzato che faciliti e migliori la produzione di software e conseguentemente 
```

anche la qualità dello stesso.

    \subsection{Qualità del prodotto software}
    Lo standard ISO/IEC:9126 definisce 6 caratteristiche centrali che influenzano la qualità di un software:
    \begin{itemize}
        \item Funzionalità
        \item Affidabilità
        \item Usabilità
        \item Efficienza
        \item Manutenibilità
        \item Portabilità
    \end{itemize}

        \subsubsection{Funzionalità}

>         Misura quanto il software soddisfa i requisiti funzionali espliciti e impliciti. Include cinque sotto-caratteristiche:

Reviewer 2: ```suggestion
        Valuta quanto il software soddisfi i requisiti funzionali espliciti e impliciti. Include cinque sotto-caratteristiche:
```

        \begin{itemize}
            \item Accuratezza - rappresenta la capacità del software di fornire i risultati concordati;
            \item Appropriatezza - il software deve offrire un appropriato insieme di funzionalità coerenti con il raggiungimento degli obiettivi prefissati all'utente;
            \item Conformità - rappresenta la capacità del prodotto di aderire agli standard e i regolamenti rilevanti nel settore in cui si opera;
            \item Interoperabilità - è la capacità del prodotto software di interagire e cooperare con uno o più sistemi esterni;

>             \item Sicurezza - indica la capacità di un prodotto software di proteggere informazioni o dati, così che tutti coloro che interagiscono con tale prodotto (sia persone che altri

Reviewer 2: ```suggestion
            \item Sicurezza - indica la capacità di un prodotto software di proteggere informazioni e dati, garantendo che tutti coloro che interagiscono con tale prodotto (sia persone che altri
```

            software) abbiano accesso alle sole informazioni che è loro consentito visualizzare secondo il livello di autorizzazione che possiedono;
        \end{itemize}

        \subsubsection{Affidabilità}
        Misura la consistenza con cui il software svolge i suoi compiti mantenendo un certo livello di prestazioni e senza andare incontro a fallimento. Include quattro sotto-caratteristiche:
        \begin{itemize}
            \item Aderenza - misura la frequenza con la quale il software è "up-and-running";
            \item Maturità - un software maturo è stato testato a fondo, pertanto molti bug sono stati individuati e risolti;
            \item Tolleranza agli errori - rappresenta la capacità del software di funzionare mantenendo determinati livelli prestazionali anche in presenza di errori;
            \item Recuperabilità - rappresenta la capacita del software di ripristinare un livello adeguato di prestazioni e di recuperare dati rilevanti a seguito di un malfunzionamento;
        \end{itemize}

>         \subsubsection{Usabilita}

Reviewer 2: ```suggestion
        \subsubsection{Usabilità}
```

            \item Comprensibilità - rappresenta la facilità con cui gli utenti possono capire le funzionalità offerte da un software;
            \item Conformità - indica la capacità del prodotto di aderire a standard e convenzioni relativi all'usabilità;
            \item Operabilità - indica quanto controllo l'utente ha sul software stesso;
        \end{itemize}

        \subsubsection{Efficienza}
        Misura le performance del sistema relativamente al quantitativo di risorse usate in determinate condizioni. L'efficienza tiene conto sia della velocità
        del sistema software che del suo uso delle risorse fisiche disponibili. Questo aspetto si può ulteriormente dividere in tre sotto-caratteristiche:
        \begin{itemize}
            \item Comportamento rispetto al tempo - è indicatore di quanto velocemente un sistema può eseguire un compito o processare una richiesta;
            \item Conformità - rappresenta la capacità del software di aderire a standard e specifiche richieste in termini di efficienza;
            \item Utilizzo delle risorse - misura quanto efficientemente un software utilizza le risorse fisiche dell'hardware sottostante;
        \end{itemize}

>         \subsubsection{Mnutenibilità}

Reviewer 2: ```suggestion
        \subsubsection{Manutenibilità}
```

    \paragraph{Test di regressione}
    Garantiscono che le modifiche apportate al codice non introducano nuovi difetti o danneggino il sistema precedentemente funzionante.\\
    Consistono nella ripetizione selettiva delle tipologie dei test elencati precedentemente e devono essere eseguiti ogni volta che si effettuano modifiche al codice.

    \paragraph{Test di accettazione}
    Sono gli ultimi test da passare prima del rilascio del software. Servono a garantire che il prodotto finale soddisfi le aspettative del proponente e degli utenti finali.

    \paragraph{Classificazione dei test}
    I test vengono identificati in base alla loro tipologia e tramite un codice numerico, specifico all'interno della categoria. I test devono avere la seguente forma:
    \begin{quote}
    \textbf{T[Tipologia Test][Codice]}\\
    \end{quote}
    Dove:
    \item [-] \textbf{Tipologia}:
    \begin{itemize}
        \item [*] \textbf{U}: Unità;
        \item [*] \textbf{I}: Integrazione;
        \item [*] \textbf{S}: Sistema;
        \item [*] \textbf{A}: Accettazione.
    \end{itemize}

    \paragraph{Stato dei test}

>     Nel \textit{Piano\_di\_Qualifica\_v1.0.0.pdf} ogni test viene seguito dal suo stato:

Reviewer 2: ```suggestion
    Nel \textit{Piano\_di\_Qualifica} ogni test viene seguito dal suo stato:
```

    \begin{itemize}
    \item \textbf{NI}: Il test non è stato implementato;
    \item \textbf{NP}: il test non è stato passato, con esito negativo;
    \item \textbf{P}: il test è stato passato, con esito positivo.
    \end{itemize}

\subsection{Gestione della Qualità}

    \subsubsection{Descrizione e Scopo}

>      La qualità viene assicurata tramite processi di verifica e validazione, che permettono di monitorare l'aderenza agli standard definiti. Una volta stabiliti i criteri qualitativi nel \textit{Piano\_di\_Qualifica\_v1.0.0.pdf}, il gruppo si impegna a seguire le best-practices per ogni fase del progetto, dai processi di sviluppo alla produzione degli artefatti. La responsabilità di garantire l'applicazione degli standard è delegata ai Verificatori, che certificano la conformità dei prodotti e dei processi agli obiettivi qualitativi.\\

Reviewer 2: ```suggestion
     La qualità viene assicurata tramite processi di verifica e validazione, che permettono di monitorare l'aderenza agli standard definiti. Una volta stabiliti i criteri qualitativi nel \textit{Piano\_di\_Qualifica}, il gruppo si impegna a seguire le best-practices per ogni fase del progetto, dai processi di sviluppo alla produzione degli artefatti. La responsabilità di garantire l'applicazione degli standard è delegata ai Verificatori, che certificano la conformità dei prodotti e dei processi agli obiettivi qualitativi.\\
```

     Il processo di gestione della qualità ha come obiettivo primario garantire che il software e i processi del ciclo di vita del progetto rispettino i requisiti specificati e siano allineati ai piani stabiliti. Attraverso l'adozione di standard qualitativi ben definiti, si mira a creare un prodotto affidabile ed efficiente, conforme alle aspettative e agli obiettivi concordati con il proponente.
    
    \subsubsection{Aspettative}
    Ci si attende che:
    \begin{itemize}
        \item Il gruppo rispetti gli standard qualitativi definiti di comune accordo;
        \item La documentazione prodotta sia chiara, coerente e conforme agli obiettivi del progetto;
        \item I prodotti software sviluppati soddisfino i requisiti funzionali e non funzionali concordati;
        \item Le pratiche di verifica e validazione siano eseguite regolarmente e con accuratezza.
    \end{itemize}
    
    \subsubsection{Metriche e Strumenti}

>     Per monitorare e valutare la qualità, vengono utilizzate specifiche metriche, adattate ai diversi processi coinvolti. Queste metriche forniscono uno strumento oggettivo per analizzare il lavoro svolto e identificare eventuali criticità. Le metriche sono descritte nel dettaglio nel \textit{Piano\_di\_Qualifica\_v1.0.0.pdf} e includono:

Reviewer 2: ```suggestion
    Per monitorare e valutare la qualità, vengono utilizzate specifiche metriche, adattate ai diversi processi coinvolti. Queste metriche forniscono uno strumento oggettivo per analizzare il lavoro svolto e identificare eventuali criticità. Le metriche sono descritte nel dettaglio nel \textit{Piano\_di\_Qualifica} e includono:
```

    \end{itemize}
    
    \subsubsection{Struttura e identificazione delle metriche}
    Le metriche adottate nel progetto sono identificate e descritte secondo una struttura standardizzata che ne facilita il riconoscimento e l'utilizzo. Ogni metrica è caratterizzata dai seguenti elementi:
    \begin{itemize}
        \item \textbf{Codice}: identificativo univoco della metrica, definito nel formato:
    \begin{quote}
        \textbf{M[abbreviazione][numero]}
    \end{quote}
    Dove:
    \begin{itemize}
        \item \textbf{[abbreviazione]}: PC se si tratta di qualità di processo, PD se si tratta di qualità di prodotto.
        \item \textbf{[numero]}: numero progressivo univoco per ciascuna metrica;
    \end{itemize}
    
    \item \textbf{Nome}: il nome completo della metrica;
    \item \textbf{Descrizione}: breve descrizione della funzionalità e dell'ambito di applicazione della metrica.
\end{itemize}

Eventualmente, le metriche possono includere:
\begin{itemize}
    \item \textbf{Formula}: modalità di calcolo della metrica.
\end{itemize}
    
    \subsubsection{Documentazione della Qualità}

>     La gestione della qualità è documentata nel \textit{Piano\_di\_Qualifica\_v1.0.0.pdf}, che definisce le metriche e le soglie accettabili per i vari processi e prodotti. Eventuali modifiche agli standard o agli strumenti vengono tracciate e aggiornate in modo continuo per garantire un processo di miglioramento costante.

Reviewer 2: ```suggestion
    La gestione della qualità è documentata nel \textit{Piano\_di\_Qualifica}, che definisce le metriche e le soglie accettabili per i vari processi e prodotti. Eventuali modifiche agli standard o agli strumenti vengono tracciate e aggiornate in modo continuo per garantire un processo di miglioramento costante.
```

    \subsubsection{Ruoli di progetto}
    Durante l'intero sviluppo del progetto, ciascun membro del gruppo assume sei ruoli distinti, ciascuno con responsabilità ben definite che riguardano specifiche attività di specifici processi.
    Ogni ruolo contribuisce al raggiungimento degli obiettivi del progetto, risultando essenziale per completare tutte le attività necessarie alla creazione di un prodotto finale di qualità.\\
    I ruoli assunti a rotazione da ciascun membro del gruppo sono i seguenti:

        \paragraph{Responsabile}
        Figura incaricata di supervisionare e coordinare le attività del gruppo, assicurandosi il raggiungimento degli obiettivi del progetto nei tempi e nei modi previsti. Si occupa di garantire una comunicazione efficace tra il gruppo e la proponente, pianifica e gestisce le risorse e valuta eventuali scelte e rischi che ne derivano.\\
        I suoi compiti sono:
        \begin{itemize}
            \item Comunicare con la proponente attraverso i canali concordati;

>             \item Stesura e avanzamento del documento \textit{Piano\_di\_Progetto\_v1.0.0.pdf};

Reviewer 2: ```suggestion
            \item Stesura e avanzamento del documento \textit{Piano\_di\_Progetto};
```

            \item Redazione dei verbali interni ed esterni;
            \item Creazione e assegnazione delle issue ai membri del team;
            \item Creazione dei diari di bordo.
        \end{itemize}

        \paragraph{Amministratore}
        Figura responsabile della creazione, manutenzione e ottimizzazione degli strumenti, delle risorse e dei processi necessari per il corretto svolgimento del progetto. Supervisiona la configurazione degli ambienti di lavoro e il monitoraggio delle scadenze amministrative, fornendo supporto al gruppo nello svolgimento delle attività.\\
        I suoi compiti sono:
        \begin{itemize}
            \item Configurazione e gestione gli ambienti di lavoro;
            \item Controllare e mantenere gli strumenti necessari per la collaborazione e la comunicazione del gruppo;
            \item Gestione del versionamento dei documenti;

>             \item Stesura e avanzamento del documento \textit{Norme\_di\_Progetto\_v1.0.0.pdf}.

Reviewer 2: ```suggestion
            \item Stesura e avanzamento del documento \textit{Norme\_di\_Progetto}.
```

        \end{itemize}

        \paragraph{Analista}
        Figura responsabile di raccogliere, analizzare e documentare i requisiti del progetto, traducendoli in casi d'uso. Collabora con la proponente per comprendere tutte le specifiche e verifica che il prodotto finale risponda alle esigenze e alle aspettative richieste.\\
        I suoi compiti sono:
        \begin{itemize}
            \item Studiare il dominio del problema e relativo contesto applicativo;
            \item Studiare i bisogni dei committenti;

>             \item Stesura del documento \textit{Analisi\_dei\_Requisiti\_v1.0.0.pdf};

Reviewer 2: ```suggestion
            \item Stesura del documento \textit{Analisi\_dei\_Requisiti};
```

>             \item Stesura e avanzamento del documento \textit{Piano\_di\_Qualifica\_v1.0.0.pdf};

Reviewer 2: ```suggestion
            \item Stesura e avanzamento del documento \textit{Piano\_di\_Qualifica};
```

        I suoi compiti sono:
        \begin{itemize}
            \item Tradurre le specifiche tecniche in codice funzionante;
            \item Creazione di test per la verifica e validazione del software;
            \item Scrittura di codice chiaro, leggibile e manutenibile;
            \item Risoluzione di eventuali bug.
        \end{itemize}

        \paragraph{Verificatore}
        Figura che si occupa di controllare che il lavoro svolto dagli altri membri del gruppo sia corretto, assicurandosi che vengano rispettate tutte le norme stabilite per il progetto.\\
        I suoi compiti sono:
        \begin{itemize}
            \item Revisionare e valutare la documentazione prodotta dai membri del gruppo;
            \item Analizzare il codice per individuare errori, discrepanze o possibili miglioramenti;
            \item Identificare e segnalare eventuali problemi.
        \end{itemize}

    \subsubsection{Cambio dei ruoli}
    Per garantire che, al completamento del progetto, ogni membro abbia ricoperto tutti i ruoli indicati, il gruppo si impegna a riassegnare i ruoli ad ogni componente del gruppo all'inizio di ogni sprint.

    \subsubsection{Tracciamento delle ore}

>     Per tracciare il tempo impiegato nei diversi ruoli durante il progetto, viene utilizzato uno spreadsheet dedicato, disponibile su Google Drive. Al termine di ogni sprint, viene utilizzato per generare il consuntivo dello sprint appena concluso e il preventivo di quello appena iniziato, infine il membro con il ruolo di Responsabile ha il compito di inserire questi elementi all'interno del documento \textit{Piano\_di\_Progetto\_v1.0.0.pdf}.

Reviewer 2: ```suggestion
    Per tracciare il tempo impiegato nei diversi ruoli durante il progetto, viene utilizzato uno spreadsheet dedicato, disponibile su Google Drive. Al termine di ogni sprint, viene utilizzato per generare il consuntivo dello sprint appena concluso e il preventivo di quello appena iniziato, infine il membro con il ruolo di Responsabile ha il compito di inserire questi elementi all'interno del documento \textit{Piano\_di\_Progetto}.
```

    La formazione risulta un pilastro fondamentale per il successo del progetto, offrendo al contempo un'opportunità di crescita professionale per tutti i partecipanti.

    \subsubsection{Formazione individuale}
    Durante lo svolgimento del progetto, ogni membro del gruppo è tenuto ad approfondire autonomamente le tecnologie necessarie per lo sviluppo del prodotto. Il processo di apprendimento è facilitato dalla condivisione di conoscenze tra i membri, grazie al contributo di coloro che possiedono maggiore esperienza e competenza.
    Questo approccio consente al gruppo di raggiungere un livello uniforme di conoscenze, adeguato al conseguimento degli obiettivi, e favorisce lo sviluppo complessivo del progetto.

\newpage
\section{Standard di progetto}

    \subsection{Standard ISO/IEC 12207 - 1995}
    \label{standard_12207}
    Lo standard ISO/IEC 12207 intitolato "Information technology – Software life cycle processes" individua delle linee guida internazionali
    per la suddivisione dei processi del ciclo di vita  di un prodotto software. Questo standard definisce un framework per tali processi

>     per tutta il corso del ciclo di vita di un prodotto software, dall'ideazione fino al ritiro.

Reviewer 2: ```suggestion
    per tutto il corso del ciclo di vita di un prodotto software, dall'ideazione fino al ritiro.
```

>     LO standard suddivide i processi in tre categorie principali che seguono la suddivisione effettuata in questo stesso documento:

Reviewer 2: ```suggestion
    Lo standard suddivide i processi in tre categorie principali che seguono la suddivisione effettuata in questo stesso documento:
```

    \begin{itemize}
        \item Processi primari
        \item Processi di supporto
        \item Processi organizzativi
    \end{itemize}

        \subsubsection{Processi primari}
        Trattasi dei processi che coinvolgono direttamente creazione, uso e supporto del software, ovvero di quei processi che descrivono

>         le attività principali del ciclo di vita del software dalla concezione dello stesso fino alla sua realizzazione. Appartenenti a questa 

Reviewer 2: ```suggestion
        le attività principali del ciclo di vita del software, dalla concezione dello stesso fino alla sua realizzazione. Appartenenti a questa 
```

        categoria sono le attività di analisi dei requisiti, progettazione, implementazione, realizzazione e esecuzione di test e infine 
        manutenzione del software.

        \subsubsection{Processi di supporto}
        Questa categoria comprende tutti i processi finalizzati a garantire che i processi primari siano svolti in maniera corretta, coerente ed

>         efficiente. Esempi concreti dki questi processi possono essere documentazione, gestione della configurazione, gestione della qualità, 

Reviewer 2: ```suggestion
        efficiente. Esempi concreti di questi processi possono essere documentazione, gestione della configurazione, gestione della qualità, 
```

                \item Interoperabilità - è la capacità del prodotto software di interagire e cooperare con uno o più sistemi esterni;
                \item Sicurezza - indica la capacità di un prodotto software di proteggere informazioni e dati, garantendo che tutti coloro che interagiscono con tale prodotto (sia persone che altri
                software) abbiano accesso alle sole informazioni che è loro consentito visualizzare secondo il livello di autorizzazione che possiedono;
            \end{itemize}

            \paragraph{Affidabilità}
            Misura la consistenza con cui il software svolge i suoi compiti mantenendo un certo livello di prestazioni e senza andare incontro a fallimento. Include quattro sotto-caratteristiche:
            \begin{itemize}
                \item Aderenza - misura la frequenza con la quale il software è "up-and-running";
                \item Maturità - un software maturo è stato testato a fondo, pertanto molti bug sono stati individuati e risolti;
                \item Tolleranza agli errori - rappresenta la capacità del software di funzionare mantenendo determinati livelli prestazionali anche in presenza di errori;

>                 \item Recuperabilità - rappresenta la capacita del software di ripristinare un livello adeguato di prestazioni e di recuperare dati rilevanti a seguito di un malfunzionamento;

Reviewer 2: ```suggestion
                \item Recuperabilità - rappresenta la capacità del software di ripristinare un livello adeguato di prestazioni e di recuperare dati rilevanti a seguito di un malfunzionamento;
```

                \item   \textbf{Formula:}
                        \[
                        ROS = \frac{RDC}{TRD} \cdot 100
                        \]
                        Dove:
                        \begin{itemize}
                            \item $RDC$: requisiti desiderabili coperti.
                            \item $TRD$: requisiti desiderabili totali.
                        \end{itemize}
            \end{itemize}
    \item   \textbf{MPD03 - Requisiti Opzionali Soddisfatti (RPS)}:
            \begin{itemize}
                \item   \textbf{Descrizione}: metrica che valuta quanti dei requisiti opzionali definiti in fase di analisi dei requisiti sono stati soddisfatti.
                \item   \textbf{Formula:}
                        \[
                        ROS = \frac{RPC}{TRP} \cdot 100
                        \]
                        Dove:
                        \begin{itemize}
                            \item $RPC$: requisiti opzionali coperti.
                            \item $TRP$: requisiti opzionali totali.
                        \end{itemize}
            \end{itemize}
    \item   \textbf{MPD04 - Function Point (FP)}:
            \begin{itemize}

>                 \item   \textbf{Descrizione}: metrica utilizzata usualmente per misurare la dimensione funzionale di un prodotto software. Si basa sulle funzionalità 

Reviewer 2: ```suggestion
                \item   \textbf{Descrizione}: metrica utilizzata per misurare la dimensione funzionale di un prodotto software. Si basa sulle funzionalità 
```

        \begin{itemize}
            \item Functional Completeness - il software deve mettere a disposizione tutte le funzionalità necessarie;
            \item Functional Correctness - il software deve eseguire le sue funzionalità correttamente e nel modo previsto;
            \item Functional Appropriateness - il software deve offrire funzionalità appropriate agli obiettivi;
        \end{itemize}

        \subsubsection{Reliability}
        Misura la consistenza con cui il software svolge i suoi compiti senza andare incontro a fallimento. Include quattro sotto-caratteristiche:
        \begin{itemize}
            \item Maturity - un software maturo è stato testato a fondo, pertanto molti bug sono stati individuati e risolti;
            \item Availability - misura la frequenza con la quale il software è "up-and-running";
            \item Fault Tolerance - rappresenta la capacità del software di funzionare anche in presenza di errori;
            \item Recoverability - rappresenta la capacita del software di recuperare da un failure riprendendo le sue funzioni;
        \end{itemize}

        \subsubsection{Usability}

>         Misura quanto intuitivo, soddisfacente da usare e user-friendly è un software. Si compone di cinque sotto-caratteristiche:

Reviewer 3: ```suggestion
        Misura quanto intuitivo, soddisfacente da usare e facile da usare è un software. Si compone di cinque sotto-caratteristiche:
```

        \begin{itemize}
            \item Understandability - rappresenta la facilità con cui gli utenti possono capire le funzionalità offerte da un software;
            \item Learnability - misura quanto facilmente un nuovo utente può imparare a fare uso di un software;
            \item Operability - indica quanto controllo l'utente ha sul software stesso;
            \item Attractiveness - misura l'attrattiva estetica che l'interfaccia del software esercita sull'utente;
            \item Accessibility - indica la facilità di uso del sistema da parte di utenti con disabilità;
        \end{itemize}

        \subsubsection{Performance Efficiency}
        Misura le performance del sistema relativamente al quantitativo di risorse usate in determinate condizioni. L'efficienza tiene conto sia della velocità

>         del sistema software che del suo uso delle risorse fisiche disponibili. Questo aspetto si puù ulteriormente dividere in tre sotto-caratteristiche:

Reviewer 3: ```suggestion
        del sistema software che del suo uso delle risorse fisiche disponibili. Questo aspetto si può ulteriormente dividere in tre sotto-caratteristiche:
```

        I suoi compiti sono:
        \begin{itemize}
            \item Tradurre le specifiche tecniche in codice funzionante;
            \item Creazione di test per la verifica e validazione del software;
            \item Scrittura di codice chiaro, leggibile e manutenibile;
            \item Risoluzione di eventuali bug;
        \end{itemize}

        \paragraph{Verificatore}
        Figura che si occupa di controllare che il lavoro svolto dagli altri membri del gruppo sia corretto, assicurandosi che vengano rispettate tutte le norme stabilite per il progetto.\\
        I suoi compiti sono:
        \begin{itemize}
            \item Revisionare e valutare la documentazione prodotta dai membri del gruppo;
            \item Analizzare il codice per individuare errori, discrepanze o possibili miglioramenti;
            \item Identificare e segnalare eventuali problemi;
        \end{itemize}

    \subsubsection{Cambio dei ruoli}

>     Per garantire che ogni membro ricopra tutti i ruoli essenziali al completamento del progetto, il gruppo si impegna a ruotarli all'inizio di ogni nuovo sprint, della durata di due settimane.

Reviewer 3: ```suggestion
    Per garantire che ogni membro ricopra tutti i ruoli essenziali al completamento del progetto, il gruppo si impegna a distribuirli all'inizio di ogni nuovo sprint, della durata di due settimane.
```
Vedi se trovi anche una parola più adatta

Author: Ho cambiato un po' la frase quindi appena carico le correzioni dagli un'occhiata

    \subsubsection{Tracciamento delle ore}
    Per tracciare il tempo impiegato nei diversi ruoli durante il progetto, viene utilizzato uno spreadsheet dedicato, disponibile su Google Drive. Al termine di ogni sprint, viene utilizzato per generare il consuntivo dello sprint appena concluso e il preventivo di quello appena iniziato, infine il membro con il ruolo di Responsabile ha il compito di inserire questi elementi all'interno del documento \textit{Piano\_di\_Progetto.pdf}.
       
        \paragraph{Preventivo}

>         Il preventivo delle ore è presentato sotto forma di tabella, che dettaglia le ore stimate per ogni membro del gruppo, suddivise per ruolo e riferite al singolo sprint. Inoltre, è incluso un diagramma circolare che mostra la ripartizione delle ore stimate per ciascun ruolo, offrendo una visione chiara e immediata delle risorse che si pensa di allocate per quello specifico sprint.

Reviewer 3: ```suggestion
        Il preventivo delle ore è presentato sotto forma di tabella, che dettaglia le ore stimate per ogni membro del gruppo, suddivise per ruolo e riferite al singolo sprint. Inoltre, è incluso un diagramma circolare che mostra la ripartizione delle ore stimate per ciascun ruolo, offrendo una visione chiara e immediata delle risorse che si pensa di allocare per quello specifico sprint.
```

        \item \textbf{Etichetta}: classifica la tipologia del compito, agevolando l'organizzazione delle issue;
        \item \textbf{Board}: indica a quale project board appartiene la issue e permette di specificare ulteriori campi, tra cui:  
        \begin{itemize}
            \item \textbf{Stato}: rappresenta l'avanzamento del compito (Todo, In Progress, Done);
            \item \textbf{Priorità}: assegna un livello di urgenza o importanza alla issue (Urgente, Importante, Moderato, Secondario);
            \item \textbf{Grandezza}: stima la complessità o il livello di sforzo richiesto per completare il compito (XS, S, M, L, XL);  
            \item \textbf{Iterazione}: specifica a quale sprint la issue è assegnata;  
        \end{itemize}  
        \item \textbf{Milestone}: identifica il traguardo o obiettivo specifico a cui la issue è associata;
    \end{itemize}
    Ogni volta che è necessario completare un compito, si segue la procedura descritta di seguito:
    \begin{enumerate}
        \item Il Responsabile crea la issue su GitHub, definendone i dettagli principali; 
        \item La issue viene assegnata ad un membro del gruppo, che si occuperà del compito;  

>         \item La issue viene associata alla milestone dello sprint attuale;  

Reviewer 3: ```suggestion
        \item La issue viene associata all'iterazione dello sprint attuale;  
```

        \item Il membro incaricato sposta la issue nello stato “In Progress” quando inizia a lavorarci;
        \item Al completamento del compito, il Verificatore designato viene informato per procedere al controllo;

>         \item Il Verificatore esamina il lavoro svolto e comunica l’esito della verifica; 

Reviewer 3: ```suggestion
        \item Il Verificatore esamina il lavoro svolto e comunica l’esito della verifica tramite GitHub; 
```
Valuta se ha senso specificarlo o meno, perché così sembra che glielo si dica a voce

        \item Se la verifica ha esito positivo, la issue viene chiusa e spostata nello stato "Done";  
    \end{enumerate}

    \subsubsection{Rischi}

>     La gestione dei rischi è un processo volto a identificare, valutare e affrontare le situazioni avverse che potrebbero influire sul successo del progetto. Questo approccio prevede l'individuazione dei rischi, la valutazione della loro gravità e occorrenza, la definizione di strategie per mitigarli, ed infine un monitoraggio costante per rilevare la possibile comparsa di nuovi rischi.\\

Reviewer 3: ```suggestion
    La gestione dei rischi è un processo volto a identificare, valutare e affrontare le situazioni avverse che potrebbero influire sul successo del progetto. Questo approccio prevede l'individuazione dei rischi, la valutazione della loro gravità e occorrenza, la definizione di strategie per mitigarli ed, infine, un monitoraggio costante per rilevare la possibile comparsa di nuovi rischi.\\
```

    I rischi vengono suddivisi in tre diverse categorie:
    \begin{itemize}
        \item [-] Rischi Tecnologici;
        \item [-] Rischi di Organizzazione;
        \item [-] Rischi di Pianificazione;
    \end{itemize}
    Ogni rischio viene classificato seguendo il formato:
    \begin{quote}
         \textbf{R[Tipo][Indice] - [Nome]}
    \end{quote}
    dove:
    \begin{itemize}
        \item \textbf{R}: Abbreviazione di "Rischio";
        \item \textbf{Tipo}: Categoria del rischio (T per Tecnologico, O per Organizzazione, P per Pianificazione);
        \item \textbf{Indice}: Identifica univocamente ciascun rischio;
        \item \textbf{Nome}: Nome del rischio;
    \end{itemize}

    \subsubsection{Coordinamento}

>     Questa attività comprende la gestione delle comunicazioni e degli incontri tra tutte le parti coinvolte nel progetto, inclusi i membri del gruppo, la proponente e i committenti.

Reviewer 3: ```suggestion
    Questa attività comprende la gestione delle comunicazioni e degli incontri tra tutte le parti coinvolte nel progetto, che comprendono i membri del gruppo, la proponente e i committenti.
```
Vedi se inserire questa frase o come cambiarla

    \subsubsection{Comunicazioni}

>     Le comunicazioni sono fondamentali per mantenere un flusso costante di informazioni tra le parti coinvolte, garantendo efficienza e coordinamento nel progetto, e supportando il buon andamento delle attività e il raggiungimento degli obiettivi prefissati.\\ 

Reviewer 3: ```suggestion
    Le comunicazioni sono fondamentali per mantenere un flusso costante di informazioni tra le parti coinvolte, garantendo efficienza e coordinamento nel progetto, supportando il buon andamento delle attività e il raggiungimento degli obiettivi prefissati.\\ 
```

    Le comunicazioni si suddividono in:
    
        \paragraph{Comunicazioni interne}

>         Le comunicazioni interne avverranno attraverso due canali principali: Telegram e Discord. Telegram, un servizio di messaggistica istantanea, sarà utilizzato per le conversazioni rapide e informali, mentre Discord, una piattaforma che supporta chat e videochiamate, sarà impiegato per le riunioni a distanza e le discussioni più complesse.\\
>         Le attività quotidiane e le comunicazioni generali si svolgeranno su Telegram, mentre le questioni critiche saranno affrontate durante degli incontri su Discord. Inoltre, su Discord è presente una sezione testuale dedicata alle comunicazioni rapide con la proponente.
>         
>         \paragraph{Comunicazioni esterne}
>         Le comunicazioni esterne avverranno tramite scambio di email, utilizzando l'indirizzo sevenbits.swe.unipd@gmail.com. Questo canale sarà utilizzato per organizzare incontri con la proponente, inviare documenti da visionare o firmare, e per risolvere eventuali dubbi.

Reviewer 3: ```suggestion
        Le comunicazioni interne avverranno attraverso due canali principali: Telegram e Discord. Telegram, un servizio di messaggistica istantanea, sarà utilizzato per le conversazioni rapide e informali, mentre Discord, una piattaforma che supporta chat e videochiamate, sarà impiegato per le riunioni a distanza e le discussioni più complesse.\\
        Le attività quotidiane e le comunicazioni generali si svolgeranno su Telegram, mentre le questioni critiche saranno affrontate durante degli incontri su Discord.
        
        \paragraph{Comunicazioni esterne}
        Le comunicazioni esterne avverranno tramite scambio di email, utilizzando l'indirizzo sevenbits.swe.unipd@gmail.com. Questo canale sarà utilizzato per organizzare incontri con la proponente, inviare documenti da visionare o firmare. Inoltre, su Discord è presente una sezione testuale dedicata alle comunicazioni rapide o dubbi con la proponente, nel loro apposito server Discord.
```
Dividere bene sezioni interne ed esterne

    \subsubsection{Riunioni}

>     Le riunioni sono fondamentali per il progresso del progetto, poiché offrono l'opportunità di confrontarsi, prendere decisioni e risolvere eventuali problematiche. Servono anche per monitorare lo stato delle attività e garantire il raggiungimento degli obiettivi prefissati.

Reviewer 3: ```suggestion
    Le riunioni sono fondamentali per il progresso del progetto, poiché offrono l'opportunità di confrontarsi, prendere decisioni e risolvere eventuali problematiche. Servono anche per monitorare lo stato di avanzamento delle attività e garantire il raggiungimento degli obiettivi prefissati.
```

    Al termine di ogni riunione, viene redatto un verbale per documentare gli argomenti trattati e le decisioni prese, garantendo così la tracciabilità e la condivisione delle informazioni. Dopo ogni incontro, il Responsabile redige il verbale, includendo tutte le informazioni rilevanti discusse, lo fa verificare e infine il Verificatore richiede l'approvazione tramite un’ulteriore verifica da parte di un altro membro. Nel caso dei verbali esterni, è prevista anche la firma della proponente.\\
    Le riunioni si distinguono in:
    
        \paragraph{Riunioni interne}

>         Le riunioni interne sono pianificate settimanalmente ogni mercoledì alle ore 16:00 o, in alternativa, subito dopo le riunioni esterne. In caso di necessità, possono essere richieste riunioni straordinarie durante la settimana, organizzate tramite un sondaggio sul canale Telegram. Tutte le riunioni si terranno sul canale Discord. I membri del gruppo si impegnano a partecipare attivamente, cercando di garantire la loro presenza. In caso di impossibilità a partecipare, ogni membro potrà recuperare la sintesi delle discussioni e le decisioni prese leggendo il verbale della riunione.

Reviewer 3: ```suggestion
        Le riunioni interne sono pianificate settimanalmente ogni mercoledì alle ore 16:00 o, in alternativa, subito dopo le riunioni esterne. In caso di necessità, possono essere richieste riunioni straordinarie durante la settimana, organizzate tramite un sondaggio sul canale Telegram. Tutte le riunioni si terranno sul canale Discord del gruppo. I membri del gruppo si impegnano a partecipare attivamente, cercando di garantire la loro presenza. In caso di impossibilità a partecipare, ogni membro potrà recuperare la sintesi delle discussioni e le decisioni prese leggendo il verbale della riunione.
```
